nixos/modules/services/hardware/nvidia-container-toolkit/cdi-generate.nix

```diff
@@ -0,0 +1,37 @@
+{ config, lib, pkgs }: let
+  mountOptions = { options = ["ro" "nosuid" "nodev" "bind"]; };
+  mounts = [
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-cuda-mps-control";
+      containerPath = "/usr/bin/nvidia-cuda-mps-control"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-cuda-mps-server";
+      containerPath = "/usr/bin/nvidia-cuda-mps-server"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-debugdump";
+      containerPath = "/usr/bin/nvidia-debugdump"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-powerd";
+      containerPath = "/usr/bin/nvidia-powerd"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-smi";
+      containerPath = "/usr/bin/nvidia-smi"; }
+    { hostPath = "${pkgs.nvidia-container-toolkit}/bin/nvidia-ctk";
+      containerPath = "/usr/bin/nvidia-ctk"; }
+    { hostPath = "${pkgs.glibc}/lib";
+      containerPath = "${pkgs.glibc}/lib"; }
```

Comment:
I forgot to ask this before. Do we not need to compute and mount the full runtime closure of these packages?

Reply:
> /lib/x86_64-linux-gnu/libc.so.6 

WOW wait! where did that come from and why does it get the higher priority than the one in the nix store?

Reply:
Yeah, they are from the container image, but in this case this can really break because they might not be compatible with the ones on NixOS. 

Let me have a look, I'll mark the PR as draft and will mark as ready again after checking.

Reply:
Hmmm which `ldd` is this using? If it's using the container's `ldd`, that's probably interpreted by `/lib/ld-linux-blahblah` instead of the nixpkgs' `ld.so`. I don't remember which priority it would assign the ld.so.cache...

But also would be nice to check the basic stuff like `echo $LD_LIBRARY_PATH`

Reply:
`LD_DEBUG=libs nvidia-smi` would also explain the selection logic

Reply:
I **think** it's fine:

<details>

<summary>ldd nvidia-smi container output</summary>

```
❯ podman run --rm --device=nvidia.com/gpu=all -it ubuntu:latest bash
root@105e28dafc7d:/# LD_DEBUG=libs nvidia-smi
        28:     find library=libpthread.so.0 [0]; searching
        28:      search path=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/glibc-hwcaps/x86-64-v3:/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/glibc-hwcaps/x86-64-v2:/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib:/nix/store/nx2c1p91xnp726v7rqxx85snd0ljkzsd-libdrm-2.4.120/lib/glibc-hwcaps/x86-64-v3:/nix/store/nx2c1p91xnp726v7rqxx85snd0ljkzsd-libdrm-2.4.120/lib/glibc-hwcaps/x86-64-v2:/nix/store/nx2c1p91xnp726v7rqxx85snd0ljkzsd-libdrm-2.4.120/lib:/nix/store/d53qm45g7yban4xwhzsss55il56ba9w1-libXext-1.3.5/lib/glibc-hwcaps/x86-64-v3:/nix/store/d53qm45g7yban4xwhzsss55il56ba9w1-libXext-1.3.5/lib/glibc-hwcaps/x86-64-v2:/nix/store/d53qm45g7yban4xwhzsss55il56ba9w1-libXext-1.3.5/lib:/nix/store/ga17mlfsss2fndyvp2s1m0s2rx2b4i9s-libX11-1.8.7/lib/glibc-hwcaps/x86-64-v3:/nix/store/ga17mlfsss2fndyvp2s1m0s2rx2b4i9s-libX11-1.8.7/lib/glibc-hwcaps/x86-64-v2:/nix/store/ga17mlfsss2fndyvp2s1m0s2rx2b4i9s-libX11-1.8.7/lib:/nix/store/wjysmbhvd706ix1hw9j1p4xpa37xjpal-libXv-1.0.12/lib/glibc-hwcaps/x86-64-v3:/nix/store/wjysmbhvd706ix1hw9j1p4xpa37xjpal-libXv-1.0.12/lib/glibc-hwcaps/x86-64-v2:/nix/store/wjysmbhvd706ix1hw9j1p4xpa37xjpal-libXv-1.0.12/lib:/nix/store/1lrxqk96vfjpf62j3xq35n01higka3kz-libXrandr-1.5.4/lib/glibc-hwcaps/x86-64-v3:/nix/store/1lrxqk96vfjpf62j3xq35n01higka3kz-libXrandr-1.5.4/lib/glibc-hwcaps/x86-64-v2:/nix/store/1lrxqk96vfjpf62j3xq35n01higka3kz-libXrandr-1.5.4/lib:/nix/store/3xk2qcmszdb3fapn7a0ys4nsxby65ir7-libxcb-1.16/lib/glibc-hwcaps/x86-64-v3:/nix/store/3xk2qcmszdb3fapn7a0ys4nsxby65ir7-libxcb-1.16/lib/glibc-hwcaps/x86-64-v2:/nix/store/3xk2qcmszdb3fapn7a0ys4nsxby65ir7-libxcb-1.16/lib:/nix/store/ilr7br1ck7i6wcgfkynwpnjp7n964h38-zlib-1.3.1/lib/glibc-hwcaps/x86-64-v3:/nix/store/ilr7br1ck7i6wcgfkynwpnjp7n964h38-zlib-1.3.1/lib/glibc-hwcaps/x86-64-v2:/nix/store/ilr7br1ck7i6wcgfkynwpnjp7n964h38-zlib-1.3.1/lib:/nix/store/1s98ricsglmfjjqkfnpvywnip5z7gp9q-gcc-13.2.0-lib/lib/glibc-hwcaps/x86-64-v3:/nix/store/1s98ricsglmfjjqkfnpvywnip5z7gp9q-gcc-13.2.0-lib/lib/glibc-hwcaps/x86-64-v2:/nix/store/1s98ricsglmfjjqkfnpvywnip5z7gp9q-gcc-13.2.0-lib/lib:/nix/store/56nk74wwpamnnwpg89vwp22kwjw5lswd-wayland-1.22.0/lib/glibc-hwcaps/x86-64-v3:/nix/store/56nk74wwpamnnwpg89vwp22kwjw5lswd-wayland-1.22.0/lib/glibc-hwcaps/x86-64-v2:/nix/store/56nk74wwpamnnwpg89vwp22kwjw5lswd-wayland-1.22.0/lib:/nix/store/q0vsspnhxy2ypy6yd3835qkv2ww4fzr7-mesa-23.3.5/lib/glibc-hwcaps/x86-64-v3:/nix/store/q0vsspnhxy2ypy6yd3835qkv2ww4fzr7-mesa-23.3.5/lib/glibc-hwcaps/x86-64-v2:/nix/store/q0vsspnhxy2ypy6yd3835qkv2ww4fzr7-mesa-23.3.5/lib:/nix/store/9l8785vc3w2jbmcvjw4gkgrszrkg103z-libGL-1.7.0/lib/glibc-hwcaps/x86-64-v3:/nix/store/9l8785vc3w2jbmcvjw4gkgrszrkg103z-libGL-1.7.0/lib/glibc-hwcaps/x86-64-v2:/nix/store/9l8785vc3w2jbmcvjw4gkgrszrkg103z-libGL-1.7.0/lib:/nix/store/h24v1xs4i9l3k18ygxvs62361rl77804-openssl-3.0.13/lib/glibc-hwcaps/x86-64-v3:/nix/store/h24v1xs4i9l3k18ygxvs62361rl77804-openssl-3.0.13/lib/glibc-hwcaps/x86-64-v2:/nix/store/h24v1xs4i9l3k18ygxvs62361rl77804-openssl-3.0.13/lib:/nix/store/xg3ld3c3xip8fx9agx901r1bz3fmnchv-dbus-1.14.10-lib/lib/glibc-hwcaps/x86-64-v3:/nix/store/xg3ld3c3xip8fx9agx901r1bz3fmnchv-dbus-1.14.10-lib/lib/glibc-hwcaps/x86-64-v2:/nix/store/xg3ld3c3xip8fx9agx901r1bz3fmnchv-dbus-1.14.10-lib/lib                (RUNPATH from file nvidia-smi)
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libpthread.so.0
        28:       trying file=/nix/store/nx2c1p91xnp726v7rqxx85snd0ljkzsd-libdrm-2.4.120/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/nx2c1p91xnp726v7rqxx85snd0ljkzsd-libdrm-2.4.120/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/nx2c1p91xnp726v7rqxx85snd0ljkzsd-libdrm-2.4.120/lib/libpthread.so.0
        28:       trying file=/nix/store/d53qm45g7yban4xwhzsss55il56ba9w1-libXext-1.3.5/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/d53qm45g7yban4xwhzsss55il56ba9w1-libXext-1.3.5/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/d53qm45g7yban4xwhzsss55il56ba9w1-libXext-1.3.5/lib/libpthread.so.0
        28:       trying file=/nix/store/ga17mlfsss2fndyvp2s1m0s2rx2b4i9s-libX11-1.8.7/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/ga17mlfsss2fndyvp2s1m0s2rx2b4i9s-libX11-1.8.7/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/ga17mlfsss2fndyvp2s1m0s2rx2b4i9s-libX11-1.8.7/lib/libpthread.so.0
        28:       trying file=/nix/store/wjysmbhvd706ix1hw9j1p4xpa37xjpal-libXv-1.0.12/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/wjysmbhvd706ix1hw9j1p4xpa37xjpal-libXv-1.0.12/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/wjysmbhvd706ix1hw9j1p4xpa37xjpal-libXv-1.0.12/lib/libpthread.so.0
        28:       trying file=/nix/store/1lrxqk96vfjpf62j3xq35n01higka3kz-libXrandr-1.5.4/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/1lrxqk96vfjpf62j3xq35n01higka3kz-libXrandr-1.5.4/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/1lrxqk96vfjpf62j3xq35n01higka3kz-libXrandr-1.5.4/lib/libpthread.so.0
        28:       trying file=/nix/store/3xk2qcmszdb3fapn7a0ys4nsxby65ir7-libxcb-1.16/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/3xk2qcmszdb3fapn7a0ys4nsxby65ir7-libxcb-1.16/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/3xk2qcmszdb3fapn7a0ys4nsxby65ir7-libxcb-1.16/lib/libpthread.so.0
        28:       trying file=/nix/store/ilr7br1ck7i6wcgfkynwpnjp7n964h38-zlib-1.3.1/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/ilr7br1ck7i6wcgfkynwpnjp7n964h38-zlib-1.3.1/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/ilr7br1ck7i6wcgfkynwpnjp7n964h38-zlib-1.3.1/lib/libpthread.so.0
        28:       trying file=/nix/store/1s98ricsglmfjjqkfnpvywnip5z7gp9q-gcc-13.2.0-lib/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/1s98ricsglmfjjqkfnpvywnip5z7gp9q-gcc-13.2.0-lib/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/1s98ricsglmfjjqkfnpvywnip5z7gp9q-gcc-13.2.0-lib/lib/libpthread.so.0
        28:       trying file=/nix/store/56nk74wwpamnnwpg89vwp22kwjw5lswd-wayland-1.22.0/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/56nk74wwpamnnwpg89vwp22kwjw5lswd-wayland-1.22.0/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/56nk74wwpamnnwpg89vwp22kwjw5lswd-wayland-1.22.0/lib/libpthread.so.0
        28:       trying file=/nix/store/q0vsspnhxy2ypy6yd3835qkv2ww4fzr7-mesa-23.3.5/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/q0vsspnhxy2ypy6yd3835qkv2ww4fzr7-mesa-23.3.5/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/q0vsspnhxy2ypy6yd3835qkv2ww4fzr7-mesa-23.3.5/lib/libpthread.so.0
        28:       trying file=/nix/store/9l8785vc3w2jbmcvjw4gkgrszrkg103z-libGL-1.7.0/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/9l8785vc3w2jbmcvjw4gkgrszrkg103z-libGL-1.7.0/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/9l8785vc3w2jbmcvjw4gkgrszrkg103z-libGL-1.7.0/lib/libpthread.so.0
        28:       trying file=/nix/store/h24v1xs4i9l3k18ygxvs62361rl77804-openssl-3.0.13/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/h24v1xs4i9l3k18ygxvs62361rl77804-openssl-3.0.13/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/h24v1xs4i9l3k18ygxvs62361rl77804-openssl-3.0.13/lib/libpthread.so.0
        28:       trying file=/nix/store/xg3ld3c3xip8fx9agx901r1bz3fmnchv-dbus-1.14.10-lib/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/xg3ld3c3xip8fx9agx901r1bz3fmnchv-dbus-1.14.10-lib/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/xg3ld3c3xip8fx9agx901r1bz3fmnchv-dbus-1.14.10-lib/lib/libpthread.so.0
        28:      search cache=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/etc/ld.so.cache
        28:      search path=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/glibc-hwcaps/x86-64-v3:/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/glibc-hwcaps/x86-64-v2:/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v3:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v2:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib               (system search path)
        28:       trying file=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/glibc-hwcaps/x86-64-v3/libpthread.so.0
        28:       trying file=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/glibc-hwcaps/x86-64-v2/libpthread.so.0
        28:       trying file=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libpthread.so.0
        28:
        28:     find library=libm.so.6 [0]; searching
        28:      search path=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib                (RUNPATH from file nvidia-smi)
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libm.so.6
        28:      search cache=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/etc/ld.so.cache
        28:      search path=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v3:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v2:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib            (system search path)
        28:       trying file=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libm.so.6
        28:
        28:     find library=libdl.so.2 [0]; searching
        28:      search path=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib                (RUNPATH from file nvidia-smi)
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libdl.so.2
        28:      search cache=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/etc/ld.so.cache
        28:      search path=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v3:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v2:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib            (system search path)
        28:       trying file=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libdl.so.2
        28:
        28:     find library=libc.so.6 [0]; searching
        28:      search path=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib                (RUNPATH from file nvidia-smi)
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libc.so.6
        28:      search cache=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/etc/ld.so.cache
        28:      search path=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v3:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v2:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib            (system search path)
        28:       trying file=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libc.so.6
        28:
        28:     find library=librt.so.1 [0]; searching
        28:      search path=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib                (RUNPATH from file nvidia-smi)
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/librt.so.1
        28:      search cache=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/etc/ld.so.cache
        28:      search path=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v3:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib/glibc-hwcaps/x86-64-v2:/nix/store/f5my15qww10swmf66ns13l24yp6j5dmq-xgcc-13.2.0-libgcc/lib            (system search path)
        28:       trying file=/nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/librt.so.1
        28:
        28:
        28:     calling init: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/ld-linux-x86-64.so.2
        28:
        28:
        28:     calling init: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libc.so.6
        28:
        28:
        28:     calling init: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/librt.so.1
        28:
        28:
        28:     calling init: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libdl.so.2
        28:
        28:
        28:     calling init: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libm.so.6
        28:
        28:
        28:     calling init: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libpthread.so.0
        28:
        28:
        28:     initialize program: nvidia-smi
        28:
        28:
        28:     transferring control: nvidia-smi
        28:
        28:     find library=libnvidia-ml.so.1 [0]; searching
        28:      search path=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib                (RUNPATH from file nvidia-smi)
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libnvidia-ml.so.1
        28:
        28:
        28:     calling init: /nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libnvidia-ml.so.1
        28:
        28:     find library=libcuda.so.1 [0]; searching
        28:      search path=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib                (RUNPATH from file nvidia-smi)
        28:       trying file=/nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libcuda.so.1
        28:
        28:
        28:     calling init: /nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libcuda.so.1
        28:
        28:
        28:     calling fini: /nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libcuda.so.1 [0]
        28:
Wed Feb 21 21:15:02 2024
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 545.29.06              Driver Version: 545.29.06    CUDA Version: 12.3     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 4090        On  | 00000000:41:00.0 Off |                  Off |
|  0%   50C    P8              14W / 450W |      3MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce RTX 2080 ...    On  | 00000000:61:00.0 Off |                  N/A |
|  0%   51C    P8              35W / 250W |      1MiB /  8192MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
        28:
        28:     calling fini:  [0]
        28:
        28:
        28:     calling fini: /nix/store/qsbx4jww0c6l5sk8c26v8pz041qmin1x-nvidia-x11-545.29.06-6.1.78/lib/libnvidia-ml.so.1 [0]
        28:
        28:
        28:     calling fini: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/librt.so.1 [0]
        28:
        28:
        28:     calling fini: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libdl.so.2 [0]
        28:
        28:
        28:     calling fini: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libm.so.6 [0]
        28:
        28:
        28:     calling fini: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libpthread.so.0 [0]
        28:
        28:
        28:     calling fini: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/libc.so.6 [0]
        28:
        28:
        28:     calling fini: /nix/store/cyrrf49i2hm1w7vn2j945ic3rrzgxbqs-glibc-2.38-44/lib/ld-linux-x86-64.so.2 [0]
        28:
```

</details>

Reply:
OK so then this must have been because of the host system's `ldd`